### PR TITLE
Pull Request for Milestone #3 - TicTacToe Game

### PR DESCRIPTION
**Pull request by tadeuasarro**
###### For this milestone we have added the following features to the project:
- Game logic;
- Check victory conditions;
- Check if the game is a draw;
- Check winning moves;
- Check if the player is choosing a valid position.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,90 +1,114 @@
...
-  break unless init != 'y' && init != 'n'
-end
 
-# Gets the information on the participants
-puts "\n\n Lets get to know our participants!!!!!!\n\n"
-player1 = ''
-player2 = ''
-init = ''
-loop do
-  print 'Enter the name of player #1: '
```
Comment (expjazz) [inline bin/main.rb]: - [x] All the game logic should be in the lib folder. Also, put every class in a separate file (inside the lib folder)

image

**Review by expjazz — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 

Great job here Team :clap: 

Your code is very creative :ok_hand: 

I loved _check_win_move_ method that lets the user know if he can win in the next move. Also kudos on checking for winners using a more algorithm approach instead of a fixed array :100: 

However, there is one big detail that you missed:

- [x] The files should be modularized, and all the logic related to the game should be in the lib folder.

- optional It would be nice if you could restart the game without exiting it.

#### For more detail, check the inline comments below.

### Please make the required changes and resubmit for a new code review once you are ready
Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

- Right now both players can have the same names. This will confuse the players when taking turns. Please put a check when the players input their names.

   image

- No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by adaorachi — APPROVED** [review]
### Hi @tadeuasarro and @mateomh :wave: 
You both did pretty well implementing the changes requested in the previous reviews. Your game behavior works well and as it should and you've met all minimum requirements for this project. Good Job! :+1: 
You can go ahead and merge this branch to the base branch of your repo while you continue working on the next milestone.

---

### STATUS: APPROVED :100: :1st_place_medal: 

---

- [**optional**] This is considered optional but it will be nice if you take some time to implement it. You should consider putting in measures to check if the player-name value inputted is an empty string and if so, disallow a user from continuing. 

Happy Coding and Good Luck in future projects :relaxed: 

MaryAnn Chukwuka

**Pull request merged**